_posts/2015-02-20-elasticsearch-and-denormalization.md

```diff
@@ -0,0 +1,223 @@
+---
+title: "ElasticSearch and Denormalization in Rails"
+layout: posts
+location: "San Francisco, CA"
+author: Nick Reavill
+author_url: 'https://www.linkedin.com/in/nreavill'
+published: true
+---
+
+Complex relational databases can lead to tortuous SQL queries and slow
+responses from your web application. If you're trying to return a long list of
+objects that are built up from five, ten or even seventeen related tables your
+response times can be unacceptably slow. Heroku web dynos time out at thirty
+seconds and if your app is regularly butting up against that limit in the course
+of normal operation then it's probably time to think about some serious
+refactoring.
+
+At Stitch Fix we have such a tool: the Receiving Queue. It is designed to give
+an itemized list of what new inventory we have received on a particular day, to
+show ordered vs received item counts and detailed information about the style
+of clothing received, so that the warehouse team can be sure they have received
+the items we were expecting.
+
+When we launched the Receiving Queue it quickly became apparent that performance
+was a problem. At first we could only load a week's worth of receipts, then a 
```

Comment:
change "receipts" to "deliveries" or: mind the terms-of-art that your average person won't know (I still get thrown by this term and it makes zero sense to me even though I know what it means).